### ISSUE TITLE
Escaped dollar sign in fixed width text macro gives "malformed argument"

### ISSUE THREAD

**Issue by brettviren** [alvinwan/TexSoup#42]
Using 0.1.3 from PyPI with Python 3.6 I find TexSoup is not parsing an escaped dollar sign inside a `\textttt{}`:

```
$ ipython
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.5.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import TexSoup

In [2]: TexSoup.TexSoup(r'\texttt{\$ foo}')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-ed0388543eb8> in <module>()
----> 1 TexSoup.TexSoup(r'\texttt{\$ foo}')

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/__init__.py in TexSoup(tex)
     82     SOUP
     83     """
---> 84     parsed, src = read(tex)
     85     return TexNode(parsed, src=src)

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/tex.py in read(tex)
     17     buf, children = Buffer(tokenize(tex)), []
     18     while buf.hasNext():
---> 19         content = read_tex(buf)
     20         if content is not None:
     21             children.append(content)

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/reader.py in read_tex(src)
    240 
    241         while src.peek() in ARG_START_TOKENS:
--> 242             expr.args.append(read_tex(src))
    243         if not expr.args:
    244             src.backward(candidate_index)

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/reader.py in read_tex(src)
    247         return expr
    248     if c in ARG_START_TOKENS:
--> 249         return read_arg(src, c)
    250     return c
    251 

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/reader.py in read_arg(src, c)
    346         else:
    347             content.append(next(src))
--> 348     return Arg.parse(content)

~/dev/memeb/venv3/lib/python3.6/site-packages/TexSoup/data.py in parse(s)
    429                             ' could not find matching punctuation for: %s.\n'
    430                             'Common issues include: Unescaped special characters,'
--> 431                             ' mistyped closing punctuation, misalignment.' % (str(s)))
    432         for arg in args:
    433             if arg.__is__(s):

TypeError: Malformed argument. First and last elements must match a valid argument format. In this case, TexSoup could not find matching punctuation for: ['{', TexCmd('')].
Common issues include: Unescaped special characters, mistyped closing punctuation, misalignment.
```

There are workarounds:

```
In [3]: TexSoup.TexSoup(r'{\texttt \$ foo}')
Out[3]: {\texttt \$ foo}
```

**Comment by SimonMaenaut** [comment]
Note: This issue was fixed by #41 .

```
simon@zap:~/TexSoup$ git log -1
commit 402bb0409b9e5ee25fea46fabe2b22b1f9e2f30c
Author: Simon Maenaut <simon@ulyssis.org>
Date:   Tue Aug 14 10:28:43 2018 +0200
    Added \( ... \) math environment (#41)
    Added \[ ... \] display math environment
    Added minor documentation and debug improvements
simon@zap:~/TexSoup$ python3.6
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from TexSoup import TexSoup
>>> tex_soup = TexSoup(r'\texttt{\$ foo}')
>>> list(tex_soup.contents)
[\texttt{\$ foo}]
>>> list(tex_soup.contents)[0].name
'texttt'
>>> list(tex_soup.texttt.contents)
['\\$', ' foo']
>>> list(tex_soup.texttt.contents)[0]
'\\$'
>>> type(list(tex_soup.texttt.contents)[0])
<class 'TexSoup.utils.TokenWithPosition'>
>>> list(tex_soup.texttt.contents)[1]
' foo'
>>> type(list(tex_soup.texttt.contents)[1])
<class 'TexSoup.utils.TokenWithPosition'>
>>> exit()
simon@zap:~/TexSoup$ 
```

**Comment by brettviren** [comment]
Also fixed for me after installing `master`.  Thanks.
